src/pages/_middleware.ts

```diff
@@ -1,5 +1,14 @@
 import { NextRequest, NextResponse } from 'next/server';
 import { rewrites } from '../../config/rewrites';
+import logger from '../utils/logger';
+
+export function logDataFromNextReqAndRes(req: NextRequest, res: NextResponse) {
+    return {
+        url: req.nextUrl.pathname,
+        method: req.method,
+        statusCode: res.status,
+    };
+}
```

Comment:
As you already pointed out in our meeting, this seems to be sort of a duplicate of `logDataFromNextAPIReqAndRes` that is in `logger.ts`. It seems the purpose of both  `logDataFromNextReqAndRes` and `logDataFromNextAPIReqAndRes` is the same but yet, there are different functions because of the different Next.js request types.